src/main/java/com/selfxdsd/selfweb/ProjectsController.java

```diff
@@ -56,12 +56,14 @@ public ProjectsController(final Self self) {
 
     /**
      * Serve a Project's page.
+     * @param provider Current Provider (Github, Gitlab etc...).
      * @param owner The owner's Username.
      * @param name The repo's name.
      * @return Project page.
      */
-    @GetMapping("/github/{owner}/{name}")
+    @GetMapping("/project/{provider}/{owner}/{name}")
```

Comment:
@criske can you remove the ``/project`` and just leave ``/{provider}/{owner}/{name}``?

I want the URLs to be as descriptive (and clean) as possible. So a Github project will have URL ``https://self-xdsd.com/github/mihai/repo`` and a GitLab project will have URL ``https://self-xdsd/gitlab/mihai/repo``.

Reply:
@amihaiemil that was my initial aproach but is not working... possible not matching the security configuration (no error is thrown on server).
 Top login buttons are gone an some images are not loading, plus jquery is not loaded.
![jquery](https://user-images.githubusercontent.com/10284893/105177653-a54fe800-5b2f-11eb-85ee-48c8f37666c4.png)


![screencapture-localhost-8080-2021-01-20-14_39_13](https://user-images.githubusercontent.com/10284893/105176566-12fb1480-5b2e-11eb-9419-062b0ec8544b.png)

Reply:
@criske And what if we have 2 identical methods in ProjectsController (returning the same html page) but with different paths? One with: ``@GetMapping(/github/{owner}/{name})`` and one with ``@GetMapping(/gitlab/{owner}/{name})``?

Reply:
@amihaiemil that works, but now adding/updating a contributor is not working anymore regardless of provider.
Also in the places where it should show "Github/Gitlab" it shows "null" (might temporary fix it if we pass the provider as model attribute still.)
But yeah, I think all of this will be properly fixed when we'll have a "provider" global variable; so maybe we should leave this change for after that?

Reply:
@criske nu inteleg, ce legatura are path-ul catre pagina de Project cu provider-ul global? :D 

Poate ai vrut sa lasi commentu asta mai jos la aia cu modelu de Repositories?

Reply:
@amihaiemil pentru ca daca pathul ar fi `/github/{owner}/{name}`, Spring nu mai vede `github` ca si variabila care poa sa fie adaugata automat ca si atribut la pagina cum sunt `owner` si `name`. (`${provider}` in pagina va fi null)
Asta rezultand  ca linia 13 din project.html

```html
<span id="owner" th:text="${owner}"/>/<span id="name" th:text="${name}"/> 
<span id="provider" th:text="${provider}" hidden/>
```
o sa bage null pe provider :D Si de acolo nu mai pot obtine providerul cu `$("#provider")` in  `getAndAddContracts.js` facand practic formularul de contract sa crape. :D

Reply:
@criske ok, inteleg acum. Si nu putem sa setam manual pe Model, exact cum e acum la Repositories? 

Deci doua metode:

```java
@GetMapping("/gitlab/{owner}/{name}")
public String gitlabProject(..., Model model) {
    model.setAttr("provider", "gitlab");
    return "project.html"
}
```
respectiv

```java
@GetMapping("/github/{owner}/{name}")
public String githubProject(..., Model model) {
    model.setAttr("provider", "github");
    return "project.html"
}
```

Reply:
@amihaiemil ba da, asta ziceam si eu mai sus: _might temporary fix it if we pass the provider as model attribute still._ :D
As sa-l adaug temporar pana cand o sa avem un provider global.

Reply:
@criske mi-a scapat, scuze. Atunci asa facem pana avem ala global :D